files/ansible/module_utils/test_result_parser/parser.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+import os
+import re
+
+from junitparser import TestCase, TestSuite, JUnitXml, Skipped, Error
+
+try:
+    from factory import TestFactory
+    from test_models import *
+except ImportError:
+    from ansible.module_utils.test_result_parser.factory import TestFactory
+    from ansible.module_utils.test_result_parser.test_models import *
+
+class TestParser(object):
+    def __init__(self, json_or_xml):
+        if str(json_or_xml) == "xml":
+            self.formatter = JUnitXml()
+            self.format = '.xml'
+        else:
+            # TODO: Implement a JSON format compatible with SQUAD
+            # This means using the python (l)xml ElementTree (<=> TestSuite())
+            self.formatter = None
+            self.format = '.json'
+
+        self.test_factory = TestFactory()
+
+    def main(self, results_path, test_threshold, output_path, extra=''):
+        result = ""
+        filename = os.path.basename(results_path)
+        with open(results_path, 'r') as output:
+            for line in output.readlines():
+                result += line
+
+        test, test_type = self.test_factory.getTest(result)
+        results, params = test.parse_output(result)
+
+        testcase = TestCase(test_type)
+
+        if test.failed(results, test_threshold):
+            testcase.result = Error(test.fail_msg.format(test_threshold), test_type)
+
+        suite = TestSuite(test_type + extra)
+        for name, value in results.items():
+            suite.add_property(name, value)
+
+        for name, value in params.items():
+            suite.add_property(name, value)
+
+        testcase.system_out = result
+        suite.add_testcase(testcase)
+
+        if self.formatter is None:
```

Comment:
this should never happen. Guarantee in the c-tor, with a raise there, and remove this check.